# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/base/nsISocketTransport.idl
# Commit: 8e7c97a37073
# Full Hash: 8e7c97a37073a14ba832e2f5c27f2c11b7787ef5
# Author: Valentin Gosu <valentin.gosu@gmail.com>
# Date: 2023-03-15 16:26:49
# Regressor Bug: 1805372
# File Overlap Count: 1
# Description:
#   Bug 1805372 - Pass effectiveTRRMode to nsHttpChannel from socketTransport/DnsAndConnectSocket r=necko-reviewers,kershaw
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D164539
# ==============================================================================

diff -r 5abfd5ad2e35 -r 8e7c97a37073 netwerk/base/nsISocketTransport.idl
--- a/netwerk/base/nsISocketTransport.idl	Wed Mar 15 12:41:26 2023 +0000
+++ b/netwerk/base/nsISocketTransport.idl	Wed Mar 15 13:00:46 2023 +0000
@@ -5,6 +5,7 @@
 
 #include "nsITransport.idl"
 #include "nsIRequest.idl"
+#include "nsITRRSkipReason.idl"
 
 interface nsIInterfaceRequestor;
 interface nsINetAddr;
@@ -346,6 +347,16 @@
     bool resolvedByTRR();
 
     /**
+     * Returns the effectiveTRRMode used for the DNS resolution.
+     */
+    readonly attribute nsIRequest_TRRMode effectiveTRRMode;
+
+    /**
+     * Returns the TRR skip reason used for the DNS resolution.
+     */
+    readonly attribute nsITRRSkipReason_value trrSkipReason;
+
+    /**
      * Indicate whether this socket is created from a private window. If yes,
      * this socket will be closed when the last private window is closed.
      */